### ISSUE TITLE
MacOS Mojave Dev Beta 8 is giving a "recursive timed mutex" error while compiling the make the file

### ISSUE THREAD

**Issue by rpranshu** [kismetwireless/kismet#80]
### **System info:**

> **MacBook Pro 2012-mid
> OS: 10.14 Beta 8 (dev)**

### **Error**

> **The Kismet module is successfully being configured but the make command is giving 4 errors**

### **Command**
```
$ make
```
### **Output**
```
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/kismet.proto
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/http.proto
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/datasource.proto
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/linuxbluetooth.proto
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/sdrrtl433.proto
protoc --cpp_out=./protobuf_cpp/ -I protobuf_definitions/ protobuf_definitions/kismet.proto
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  protobuf_cpp/kismet.pb.cc | sed -e "s/\.o/\.cc.o/" > protobuf_cpp/kismet.pb.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c protobuf_cpp/kismet.pb.cc -o protobuf_cpp/kismet.pb.cc.o
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  protobuf_cpp/http.pb.cc | sed -e "s/\.o/\.cc.o/" > protobuf_cpp/http.pb.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c protobuf_cpp/http.pb.cc -o protobuf_cpp/http.pb.cc.o
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  protobuf_cpp/datasource.pb.cc | sed -e "s/\.o/\.cc.o/" > protobuf_cpp/datasource.pb.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c protobuf_cpp/datasource.pb.cc -o protobuf_cpp/datasource.pb.cc.o
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  protobuf_cpp/linuxbluetooth.pb.cc | sed -e "s/\.o/\.cc.o/" > protobuf_cpp/linuxbluetooth.pb.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c protobuf_cpp/linuxbluetooth.pb.cc -o protobuf_cpp/linuxbluetooth.pb.cc.o
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  protobuf_cpp/sdrrtl433.pb.cc | sed -e "s/\.o/\.cc.o/" > protobuf_cpp/sdrrtl433.pb.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c protobuf_cpp/sdrrtl433.pb.cc -o protobuf_cpp/sdrrtl433.pb.cc.o
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -MM -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  util.cc | sed -e "s/\.o/\.cc.o/" > util.cc.d
g++ -std=gnu++11 -std=gnu++14 -std=gnu++17 -Wall -g -I. -g -O2 -O3 -pthread -I/usr/local/Cellar/protobuf/3.6.0/include -I/opt/local/include  -c util.cc -o util.cc.o
In file included from util.cc:66:
In file included from ./packet.h:36:
In file included from ./globalregistry.h:29:
./kis_mutex.h:119:37: error: no member named 'native_handle' in
      'std::__1::recursive_timed_mutex'
            auto native_m = cpplock.native_handle();
                            ~~~~~~~ ^
./kis_mutex.h:134:37: error: no member named 'native_handle' in
      'std::__1::recursive_timed_mutex'
            auto native_m = cpplock.native_handle();
                            ~~~~~~~ ^
./kis_mutex.h:186:38: error: no member named 'native_handle' in
      'std::__1::recursive_timed_mutex'
            auto native_m = cpplock->native_handle();
                            ~~~~~~~  ^
./kis_mutex.h:212:33: error: no member named 'native_handle' in
      'std::__1::recursive_timed_mutex'
            auto native_m = in->native_handle();
                            ~~  ^
4 errors generated.
make: *** [util.cc.o] Error 1
``` 
@kismetwireless 

### **Additional Info**

The ```./configure``` command executed successfully
```
Configuration complete: 
         Compiling for: darwin18.0.0 (x86_64)
           C++ Library: stdc++
   Installing as group: wheel
    Man pages owned by: 
       Installing into: /usr/local
          Setuid group: staff
   Linux WEXT capture : n/a (only Linux)
        Prelude  SIEM : no
   PCRE Regex Filters : yes
          pcap capture: yes
       airpcap control: n/a (only Cygwin/Win32)
        PPI log format: yes
LibCapability (enhanced
   privilege dropping): n/a (only Linux)
   OSX/Darwin capture : yes
         Linux Netlink: n/a (only Linux)
  Linux NetworkManager: n/a (only Linux)
       Linux Bluetooth: n/a (only Linux)
        Python Modules: yes
       RTL-SDR RTL_433: yes
      Freaklabs Zigbee: yes
 lm-sensors monitoring: n/a (only Linux)
        Built-in Debug: partial - Missing libdw or libbfd will not print full stacks on crash
```

**Comment by kismetwireless** [comment]
This should no longer be the case in the current git versions; the native handle is no longer used for debugging.

**Comment by rpranshu** [comment]
Then what is the solution?

**Comment by kismetwireless** [comment]
The solution is to use the git-master code which does not include references to native thread handles (and hasn't since early august)

**Comment by rpranshu** [comment]
### **Somewhere in between the ```make``` process i get the following warnings (3):**

```
In file included from kis_dlt_radiotap.cc:53:
In file included from ./kis_datasource.h:38:
In file included from ./protobuf_cpp/kismet.pb.h:9:
In file included from /usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/common.h:46:
/usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/port.h:379:9: warning: 'bswap_16' macro redefined
      [-Wmacro-redefined]
#define bswap_16(x) OSSwapInt16(x)
        ^
./local_ieee80211_radiotap.h:21:9: note: previous definition is here
#define bswap_16 OSSwapInt16
        ^
In file included from kis_dlt_radiotap.cc:53:
In file included from ./kis_datasource.h:38:
In file included from ./protobuf_cpp/kismet.pb.h:9:
In file included from /usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/common.h:46:
/usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/port.h:380:9: warning: 'bswap_32' macro redefined
      [-Wmacro-redefined]
#define bswap_32(x) OSSwapInt32(x)
        ^
./local_ieee80211_radiotap.h:22:9: note: previous definition is here
#define bswap_32 OSSwapInt32
        ^
In file included from kis_dlt_radiotap.cc:53:
In file included from ./kis_datasource.h:38:
In file included from ./protobuf_cpp/kismet.pb.h:9:
In file included from /usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/common.h:46:
/usr/local/Cellar/protobuf/3.6.0/include/google/protobuf/stubs/port.h:381:9: warning: 'bswap_64' macro redefined
      [-Wmacro-redefined]
#define bswap_64(x) OSSwapInt64(x)
        ^
./local_ieee80211_radiotap.h:23:9: note: previous definition is here
#define bswap_64 OSSwapInt64
        ^
3 warnings generated.
```

### **Also I get 2 Errors regarding ```sys/signalfd.h```**
```
gcc -MM -Wall -g    -pthread -I/usr/local/Cellar/protobuf-c/1.3.1/include  capture_nrf_mousejack.c | sed -e "s/\.o/\.c.o/" > capture_nrf_mousejack.c.d
capture_nrf_mousejack.c:36:10: fatal error: 'sys/signalfd.h' file not found
#include <sys/signalfd.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
gcc -Wall -g    -pthread -I/usr/local/Cellar/protobuf-c/1.3.1/include  -I/usr/local/Cellar/libusb/1.0.22/include/libusb-1.0 -c capture_nrf_mousejack.c -o capture_nrf_mousejack.c.o
capture_nrf_mousejack.c:36:10: fatal error: 'sys/signalfd.h' file not found
#include <sys/signalfd.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [capture_nrf_mousejack.c.o] Error 1
make: *** [capture_nrf_mousejack/kismet_cap_nrf_mousejack] Error 2
```
